File: Cleaning Product Names.R
clean_product_names <- function(data, name_column) {
  data[[name_column]] <- gsub("[[:punct:]]", "", data[[name_column]])
  data[[name_column]] <- gsub("[][!#$%()*,.:;<=>@^_`|~.{}]", "", data[[name_column]])
  data[[name_column]] <- gsub("[[:digit:]]+", "", data[[name_column]])
  data[[name_column]] <- toupper(data[[name_column]])
  data[[name_column]] <- trimws(data[[name_column]])
  data[[name_column]] <- gsub(' +', ' ', data[[name_column]])
  return(data)
}

File: Cleaning Product Names.py
def clean_product_names(data, name_column):
    data[name_column] = data[name_column].str.replace(r'[^\w\s]', '', regex=True)
    data[name_column] = data[name_column].str.upper()
    data[name_column] = data[name_column].str.strip()
    return data
